File: ruby/difference-of-squares/difference_of_squares.rb
class Squares
  def initialize(n)
    @series = (1..n)
  end

  def square_of_sum
    series.sum**2
  end

  def sum_of_squares
    series.map { |i| i**2 }.sum
  end

  def difference
    square_of_sum - sum_of_squares
  end

  private

  attr_reader :series
end

File: go/difference-of-squares/difference_of_squares.go
package diffsquares

func SumOfSquares(n int) int {
	result := 0

	for i := 1; i <= n; i++ {
		result += i * i
	}

	return result
}

func SquareOfSum(n int) int {
	result := 0

	for i := 1; i <= n; i++ {
		result += i
	}

	return result * result
}

func Difference(n int) int {
	return SquareOfSum(n) - SumOfSquares(n)
}
